File: frontend/src/types/FileAnnotationResponse.ts
export interface FileAnnotationResponse {
    fileId: number;
    fileName: string;
    fileUrl: string;               // ← 이걸 PDFViewer에 던질 거야
    snapshotCreatedAt: string;
    version: number;
    latest: boolean;
    slides: {
      pageNumber: number;
      annotations: Array<{
        id: string;
        text?: string;
        payload?: Record<string, any>;
        position: { x: number; y: number };  // 0~1
        size: { width: number; height: number }; // 0~1
        source: 'PPT' | 'MANUAL' | 'PDF';
        order?: number;
        createdAt?: string;
        updatedAt?: string;
        answerState?: number;
      }>;
    }[];
  }

File: backend/src/main/java/org/example/speaknotebackend/dto/FileAnnotationResponse.java
package org.example.speaknotebackend.dto;

import java.time.Instant;
import java.util.List;
import java.util.Map;
import lombok.*;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class FileAnnotationResponse {
    // 파일 메타
    private Long fileId;
    private String fileName;
    private String fileUrl;          // 바로 PDF 뷰어에 넣으면 됨
    private Instant snapshotCreatedAt;

    // 버전
    private Integer version;
    private boolean latest;

    // 주석
    private List<SlideDto> slides;

    @Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
    public static class SlideDto {
        private int pageNumber;
        private List<AnnoDto> annotations;
    }

    @Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
    public static class AnnoDto {
        private String id;
        private String text;
        private Map<String, Object> payload;
        private Pos position; // [0,1]
        private Size size;    // [0,1]
        private String source; // PPT | MANUAL | PDF
        private Integer order;
        private Instant createdAt;
        private Instant updatedAt;
        private Integer answerState;
    }

    @Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
    public static class Pos { private double x; private double y; }

    @Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
    public static class Size { private double width; private double height; }
}
